File: Problems/0458_Poor_Pigs/Poor_Pigs.py
# coding: utf-8

import os
import sys
import time

class Solution:
    def poorPigs(self, buckets, minutesToDie, minutesToTest):
        pigs = 0
        while (minutesToTest / minutesToDie + 1) ** pigs < buckets:
            pigs += 1
        return pigs

def main():
    argv = sys.argv
    argc = len(argv)

    if (argc < 2):
        print("Usage: python %s <testdata.txt>" %(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("%s not found..." %argv[1])
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        print("argv[1] = %s" %temp)
        loop_main(temp)
    #    print("Hit Return to continue...")
    #    input()

def loop_main(temp):
    flds = temp.replace("[","").replace("]","").rstrip().split(",")
    buckets = int(flds[0])
    minutesToDie = int(flds[1])
    minutesToTest = int(flds[2])

    time0 = time.time()

    sl = Solution()
    result = sl.poorPigs(buckets, minutesToDie, minutesToTest)

    print("result = %s" %result)

    time1 = time.time()
    print("Execute time ... : %f[s]" %(time1 - time0))
    print()

if __name__ == "__main__":
    main()

File: Problems/0458_Poor_Pigs/Poor_Pigs.cs
using System;

public class Solution
{
    public int PoorPigs(int buckets, int minutesToDie, int minutesToTest)
    {
        int pigs = 0;
        while (checked(Math.Pow(minutesToTest / minutesToDie + 1, pigs)) < buckets)
            pigs++;
        return pigs;
    }

    public void Main(string args)
    {
        string[] flds = args.Replace("[","").Replace("]","").Trim().Split(',');
        int buckets = int.Parse(flds[0]);
        int minutesToDie = int.Parse(flds[1]);
        int minutesToTest = int.Parse(flds[2]);

        if (minutesToDie > minutesToTest)
        {
            Console.WriteLine("minutesToTest is lower than minutesToTest.");
            return;
        }

        Console.WriteLine("buckets = " + buckets.ToString() + ", minutesToDie = " + minutesToDie.ToString() + ", minutesToTest = " + minutesToTest.ToString());
        
        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();

        int result = PoorPigs(buckets, minutesToDie, minutesToTest);
        Console.WriteLine("result = " + result.ToString());
        
        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}
